airflow_kjo/kubernetes_job_launcher.py

```diff
@@ -84,40 +83,37 @@ def _tail_pod_logs(self, name, namespace, job):
             # https://raw.githubusercontent.com/kubernetes-client/python/master/kubernetes/client/api/core_v1_api.py
             if bool(num_lines):
                 read_log = self.kube_pod_client.read_namespaced_pod_log(
-                    name=pod_name, namespace=namespace, tail_lines=num_lines
+                    name=pod_name, namespace=self.namespace, tail_lines=num_lines
                 )
                 line_or_lines = "line" if num_lines == 1 else "lines"
-                msg = f'Reading last {num_lines} {line_or_lines} from log for Pod "{pod_name}" in Namespace "{namespace}"'
+                msg = f'Reading last {num_lines} {line_or_lines} from log for Pod "{pod_name}" in Namespace "{self.namespace}"'
             else:
-                msg = f'Reading full logfile for Pod "{pod_name}" in Namespace "{namespace}"'
+                msg = f'Reading full logfile for Pod "{pod_name}" in Namespace "{self.namespace}"'
                 # could this ever be too much data?
                 read_log = self.kube_pod_client.read_namespaced_pod_log(
-                    name=pod_name, namespace=namespace
+                    name=pod_name, namespace=self.namespace
                 )
             lines = [line for line in read_log]
             str_lines = "".join(lines).strip()
             if str_lines:
                 logging.info(msg)
                 logging.info(f"Reading....\n{str_lines}")
-                logging.info(f'End log for Pod "{pod_name}" in Namespace "{namespace}"')
+                logging.info(f'End log for Pod "{pod_name}" in Namespace "{self.namespace}"')
                 had_logs = True
         return had_logs
 
-    @staticmethod
-    def _expand_yaml_obj_with_configuration(yaml_obj, configuration, overwrite=False):
+    def _expand_yaml_obj_with_configuration(self, configuration, overwrite=False):
```

Comment:
This was better with the yaml_obj over self because self is related to the entire instance but this only depends on the yaml_obj.

Reply:
good point